api/utils/db.ts

```diff
@@ -0,0 +1,62 @@
+import { connect, connection, disconnect } from 'mongoose'
+import { MongoMemoryServer } from 'mongodb-memory-server'
+import config from '../src/config'
+import logger from '../utils/logger'
+class MongoDBConnection {
+    private static _instance: MongoDBConnection
+    private _mongoServer?: MongoMemoryServer
```

Comment:
Raczej nie lubię dodawania `_` jako prefix dla prywatnych pól. Jeśli już miałbym jakieś prefixy stosować to raczej bym się skłonił do stylu ES i dawał `#` (tylko wtedy musi zniknąć TSowy `private`)